vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewManager.java

```diff
@@ -0,0 +1,123 @@
+
+        Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap = new HashMap<>();
+        Iterable<SpaceLinkDisplay> spaceToSpaceLinksList = spaceLinkDisplayRepository.findAll();
+        for (SpaceLinkDisplay link : spaceToSpaceLinksList) {
+            if (link.getLink() != null && link.getLink().getSourceSpace() != null) {
+                String spaceId = link.getLink().getSourceSpace().getId();
+                if (!spaceToSpaceLinksMap.containsKey(spaceId)) {
+                    spaceToSpaceLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToSpaceLinksMap.get(spaceId).add(link);
+            }
+        }
+        return spaceToSpaceLinksMap;
+    }
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space(i.e. a space connected with all other spaces and modules)
+     * 
+     * @param spaceLinkMap          For holding list of all links corresponding to
+     *                              each space
+     * @param spaceToModuleLinksMap For getting list of links from one space to
+     *                              other modules
+     * @param spaceToSpaceLinksMap  For getting list of links from one space to
+     *                              other spaces
+     */
+    @Override
+    public Map<String, List<String>> getSpaceLinkedToSpacesAndModules(
+            Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap,
+            Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap) {
+
+        Map<String, List<String>> spaceLinkMap = new LinkedHashMap<>();
+        Iterable<Space> allSpacesList = spaceRepo.findAll();
+        if (allSpacesList != null) {
+            for (Space space : allSpacesList) {
+                List<ModuleLinkDisplay> spaceToModulelinksList = spaceToModuleLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceToSpaceLinksList = spaceToSpaceLinksMap.get(space.getId());
+                List<String> listOfSpaceIdAndModuleIds = new ArrayList<>();
+
+                if (spaceToModulelinksList != null) {
+                    spaceToModulelinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getModule() != null) {
+                            listOfSpaceIdAndModuleIds.add(link.getLink().getModule().getId());
+                        }
+                    });
+                }
+                if (spaceToSpaceLinksList != null) {
+                    spaceToSpaceLinksList.forEach(link -> {
+                        if (link.getLink() != null && link.getLink().getTarget() != null) {
+                            listOfSpaceIdAndModuleIds.add(link.getLink().getTarget().getId());
+                        }
+                    });
+                }
+                spaceLinkMap.put(space.getId(), listOfSpaceIdAndModuleIds);
+            }
+        }
+        return spaceLinkMap;
+    }
+}
```

Comment:
this class still seems to mix generating the data structure for the map and then also turning that into JSON. A class should only have one purpose (Single Responsibility Principle). So this class should generate the data structure and the jsonhelper should generate the json from it.